code/model/editableformfields/EditableFormField.php

```diff
@@ -335,20 +345,30 @@ public function canDelete($member = null) {
 	}
 
 	/**
-	 * Return whether a user can edit this form field
-	 * based on whether they can edit the page
+	 * Return whether a user can view this form field
+	 * based on whether they can view the page
 	 *
 	 * @return bool
 	 */
-	public function canEdit($member = null) {
+	public function canView($member = null) {
 		if($this->Parent()) {
-			return $this->Parent()->canEdit($member) && !$this->isReadonly();
+			return $this->Parent()->canView($member) && !$this->isReadonly();
 		}
 
 		return true;
 	}
 
 	/**
+	 * Return whether a user can create this form field
+	 * based on whether they can edit the page
+	 *
+	 * @return bool
+	 */
+	public function canCreate($member = null) {
```

Comment:
canCreate will be called on singletons, and thus canEdit will always return true (because of the parent check).

You will need to use the $context parameter to pass in the parent (how SiteTree implements canView).